internal/command/testdata/test/pass_var_inside_variables/main.tftest.hcl

```diff
@@ -0,0 +1,10 @@
+variables {
+  var2 = var.var1 ? "true" : "false"
+}
```

Comment:
I'd suggest to add another test - We'd want one test checking an explicit value, and one test checking a default value

Reply:
sure, I have the added the test for default value in latest [commit](https://github.com/opentofu/opentofu/pull/1488/commits/0777ea9d965f9269039da0ad55654637576eb30d)